backend/src/main/java/com/bakdata/conquery/models/query/queryplan/specific/ExternalNode.java

```diff
@@ -69,7 +72,7 @@ public void nextTable(QueryExecutionContext ctx, Table currentTable) {
 
 		if (table.equals(currentTable) && contained != null){
 			dateUnion.addAll(contained);
-			dateUnion.retainAll(ctx.getDateRestriction());
+			dateUnion.retainAll(dateRestriction);
```

Comment:
Warum reicht hier nicht die Restriction aus dem QueryExecutionContext?

Reply:
Es gibt keine, weil ExternalNode keine ValidityDateNode ist und sie somit nicht gesetzt ist

Reply:
Ich habe nicht verstanden warum DateRestriction und ValidityDate so sehr verstrickt sein mussten, deswegen habe ich es mal aus einander gezogen. War das ein Performance Ding? https://github.com/ingef/conquery/pull/2963

Reply:
Ich hab keine Ahnung warum das so implementiert ist. Ich hätte denke ich ValidityDateNode in die ConceptNode reingezogen aber ich hab den Code auch nicht so ganz verstanden.